discord/ext/commands/cog.py

```diff
@@ -505,9 +521,14 @@ def _inject(self, bot: BotBase, override: bool, guild: Optional[Snowflake], guil
                 # This is already atomic
                 bot.tree.add_command(command, override=override, guild=guild, guilds=guilds)
 
+        try:
+            await maybe_coroutine(self.cog_load)
+        except Exception:
+            pass
```

Comment:
Do we want errors on load to propagate? we don't do this for unload.

Reply:
If this is going to be propagated it needs to be in the beginning of the method rather than at the end as it is here. I think propagating is fine since `__init__` propagates. Unload doesn't need to because it's a finalizer, an error that occurs during finalisation is pretty much can't be cleanly handled anyway.